File: packages/xpce/prolog/lib/compatibility/global.pl
/*  $Id$

    Part of XPCE
    Designed and implemented by Anjo Anjewierden and Jan Wielemaker
    E-mail: jan@swi.psy.uva.nl

    Copyright (C) 1992 University of Amsterdam. All rights reserved.
*/

:- module_transparent
	new_global/2.


new_global(Ref, Term) :-
	new(Ref, Term),
	send(Ref, lock_object, @on).

File: packages/xpce/CXX/demo/global.cxx
/*  $Id$

    Part of XPCE
    Designed and implemented by Anjo Anjewierden and Jan Wielemaker
    E-mail: jan@swi.psy.uva.nl

    Copyright (C) 1994 University of Amsterdam. All rights reserved.
*/

/* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Very simple demo, illustrating a single global C++ object referencing an
XPCE object.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - */

#include <pce/Pce.h>
#include <pce/Picture.h>

PcePicture p("Hello World");

PceStatus
pceInitApplication(int argc, char *argv[])
{ p.send("open");

  return SUCCEED;
}
